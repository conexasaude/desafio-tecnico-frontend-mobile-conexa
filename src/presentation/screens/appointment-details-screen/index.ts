export * from './appointment-details-screen';
