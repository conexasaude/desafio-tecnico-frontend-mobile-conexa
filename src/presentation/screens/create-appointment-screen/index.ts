export * from './create-appointment-screen';
