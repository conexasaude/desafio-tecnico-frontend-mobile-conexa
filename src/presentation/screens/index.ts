export * from './appointments-screen';
export * from './create-appointment-screen';
export * from './login-screen';
