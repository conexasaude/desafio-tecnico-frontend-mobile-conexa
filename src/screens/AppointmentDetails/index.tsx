import { View } from 'react-native';
import { NativeStackScreenProps } from "@react-navigation/native-stack";
import { FontAwesome } from '@expo/vector-icons';
import RootStackParamList from '../../types/rootStackParamList';
import formatsDate from '../../utils/formatsDate';
import { Container, Content, Field, FieldContainer, Header, FieldText, Label } from './styles';
import { useEffect } from 'react';

type Props = NativeStackScreenProps<RootStackParamList, 'AppointmentDetails'>;

export default function AppointmentDetails({ route, navigation }: Props) {
	const { appointment } = route.params;
	
	useEffect(() => {
		navigation.setOptions({ title: `Consulta ${appointment.id}` })
	}, [])

	return (
		<Container>
			<Header>
				<View>
					<FontAwesome name="user-circle-o" size={72} color="black" />
				</View>
				<FieldContainer>
					<Field>
						<Label>Paciente:</Label>
						<FieldText>{appointment.paciente}</FieldText>
					</Field>

					<Field>
						<Label>Data e hora: </Label>
						<FieldText>{`${formatsDate(appointment.dataConsulta, 'weekday-date-time')}`}</FieldText>
					</Field>
				</FieldContainer>
			</Header>
				
			<Content>
				<Field>
					<Label>Observação: </Label>
					<FieldText>{appointment.observacao}</FieldText>
				</Field>
			</Content>
		</Container>
    )
}