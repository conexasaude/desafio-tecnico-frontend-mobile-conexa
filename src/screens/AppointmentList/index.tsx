import { FlatList, View } from 'react-native';
import { NativeStackScreenProps } from "@react-navigation/native-stack";
import { useEffect, useState } from 'react';
import Spinner from 'react-native-loading-spinner-overlay';
import { getAppointments } from '../../services/appointment';
import { Appointment } from '../../types/appointment';
import RootStackParamList from '../../types/rootStackParamList';
import AppointmentCard from '../../components/AppointmentCard';
import SearchInput from '../../components/SearchInput';
import { Container } from './styles';

type Props = NativeStackScreenProps<RootStackParamList, 'AppointmentList'>;

export default function AppointmentList({ navigation }: Props) {
	const [appointments, setAppointments] = useState<Appointment[]>([])
	const [loading, setLoading] = useState(false)
	const [search, setSearch] = useState("")

	async function fetchAppointments() {
		setLoading(true)
		const response = await getAppointments()
		setAppointments(response)
		setLoading(false)
	}

	async function navigateToDetails(appointment: Appointment) {
		navigation.navigate('AppointmentDetails', {
			appointment
		})
	}

	function filterAppointments(appointments: Appointment[]) {
		if(search !== '') {
			return appointments.filter(appointment => 
				String(appointment.id).toLocaleLowerCase().includes(search.toLocaleLowerCase())  || appointment.paciente.toLocaleLowerCase().includes(search.toLocaleLowerCase())
			)
		} else {
			return appointments
		}
	}
    
	useEffect(() => {
		fetchAppointments()
	}, [])

	return (
		<>
			<Spinner
				visible={loading}
				textContent={'Carregando...'}
				textStyle={{color: '#FFF'}}
			/>
				
			<Container>
				<SearchInput value={search} onChangeText={setSearch} />

				<FlatList
					data={filterAppointments(appointments)}
					renderItem={({item}) => 
						<AppointmentCard
							id={String(item.id)}
							onPress={() => navigateToDetails(item)} 
							pacient={item.paciente}
							date={item.dataConsulta}
						/>
					}
					style={{ paddingHorizontal: 16 }}
				>
				</FlatList>
			</Container>
		</>
	)
}