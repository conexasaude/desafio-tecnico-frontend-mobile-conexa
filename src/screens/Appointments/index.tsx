import React, { useEffect } from 'react';
import { useReduxSelector } from '@/hooks/useReduxSelector';
import { useReduxDispatch } from '@/hooks/useReduxDispatch';
import { FlatList } from 'react-native';
import { Card } from '@/components/Card';
import { useTheme } from '@emotion/react';
import { useNavigation } from '@react-navigation/native';
import { NavPropsLogged } from '@/routes/logged';
import { fetchAppointments } from '@/store/thunks/appointment.thunk';
import { useToast } from 'react-native-toast-notifications';
import * as Styled from './styles';

export function AppointmentsScreen() {
  const { show } = useToast();
  const { effects } = useTheme();
  const { navigate } = useNavigation<NavPropsLogged>();
  const { appointments } = useReduxSelector(state => state.appointment);

  const dispatch = useReduxDispatch();

  useEffect(() => {
    dispatch(fetchAppointments({ onError: show }));
  }, [show, dispatch]);

  return (
    <Styled.SafeArea>
      <Styled.Container>
        <Styled.Column>
          <Styled.Title>Painel de Consultas</Styled.Title>
          <Styled.Line />
        </Styled.Column>

        <Styled.Row>
          <Styled.DescriptionList>{`${appointments.length} consulta(s) agendadas`}</Styled.DescriptionList>
          <Styled.ScheduleButton
            title="Agendar Consulta"
            onPress={() => navigate('ScheduleAppointment')}
          />
        </Styled.Row>

        <FlatList
          data={appointments}
          keyExtractor={({ id }) => id.toString()}
          renderItem={({ item }) => (
            <Card
              testID="appointment-card"
              title={item.paciente}
              description={item.dataConsulta}
              button={{
                title: 'Ver Detalhes',
                onPress: () => navigate('DetailsAppointment', { id: item.id }),
              }}
            />
          )}
          contentContainerStyle={{ paddingHorizontal: effects.spacing.md }}
        />
      </Styled.Container>
    </Styled.SafeArea>
  );
}
