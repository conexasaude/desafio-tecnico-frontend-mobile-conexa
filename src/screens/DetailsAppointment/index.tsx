import React from 'react';
import * as Styled from './styles';

export function DetailsAppointmentScreen() {
  return (
    <Styled.Container>
      <Styled.Title>Details Appointment Screen</Styled.Title>
    </Styled.Container>
  );
}
