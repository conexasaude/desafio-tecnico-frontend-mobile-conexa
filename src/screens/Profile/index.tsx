import React from 'react';
import Icon from 'react-native-vector-icons/MaterialCommunityIcons';
import * as Styled from './styles';
import { useTheme } from '@emotion/react';
import { useReduxSelector } from '@/hooks/useReduxSelector';
import { useReduxDispatch } from '@/hooks/useReduxDispatch';
import { userSlice } from '@/store/slices/user.slice';

export function ProfileScreen() {
  const { colors } = useTheme();
  const { user } = useReduxSelector(state => state.user);
  const dispatch = useReduxDispatch();

  return (
    <Styled.SafeArea>
      <Styled.Container>
        <Icon name="account-circle" size={64} color={colors.gray[400]} />
        <Styled.LabelName>{user?.nome}</Styled.LabelName>
        <Styled.LabelEmail>{user?.email}</Styled.LabelEmail>

        <Styled.Spacer />

        <Styled.LoggoutButton
          variant="outlined"
          title="Sair"
          onPress={() => dispatch(userSlice.actions.loggout())}
        />
      </Styled.Container>
    </Styled.SafeArea>
  );
}
