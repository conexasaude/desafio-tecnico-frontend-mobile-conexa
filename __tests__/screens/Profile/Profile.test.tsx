import React from 'react';
import {
  renderWithThemeProvider,
  matchSnapshotWithProvider,
} from '@/lib/functions/Testing';

import { ProfileScreen } from '@/screens/Profile';
import { fireEvent } from '@testing-library/react-native';

const mockDispatch = jest.fn();

jest.mock('@/hooks/useReduxDispatch', () => ({
  useReduxDispatch: () => mockDispatch,
}));

jest.mock('@/hooks/useReduxSelector', () => ({
  useReduxSelector: () => ({ loading: false }),
}));

describe('ProfileScreen', () => {
  test('should render correctly', () => {
    renderWithThemeProvider(<ProfileScreen />);
  });

  test('should matches snapshot', () => {
    matchSnapshotWithProvider(<ProfileScreen />);
  });

  test('should call logout if press "sair"', async () => {
    const { getByText } = renderWithThemeProvider(<ProfileScreen />);
    const logoutButton = getByText('Sair');
    fireEvent.press(logoutButton);
    expect(mockDispatch).toHaveBeenCalled();
  });
});
