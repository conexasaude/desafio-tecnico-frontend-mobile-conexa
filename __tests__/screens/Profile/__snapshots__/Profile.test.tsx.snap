// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProfileScreen should matches snapshot 1`] = `
<RCTSafeAreaView
  style={
    [
      {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    style={
      [
        {
          "alignItems": "center",
          "flex": 1,
          "paddingHorizontal": 16,
          "paddingVertical": 32,
        },
        undefined,
      ]
    }
  >
    <Text
      allowFontScaling={false}
      selectable={false}
      style={
        [
          {
            "color": "#B3B3B3",
            "fontSize": 64,
          },
          undefined,
          {
            "fontFamily": "Material Design Icons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          {},
        ]
      }
    >
      ó°€‰
    </Text>
    <Text
      style={
        [
          {
            "color": "#4C4C4C",
            "fontFamily": "Nunito-Bold",
            "fontSize": 16,
            "marginBottom": 4,
            "marginTop": 16,
          },
          undefined,
        ]
      }
    />
    <Text
      style={
        [
          {
            "color": "#7F7F7F",
            "fontFamily": "Nunito-SemiBold",
            "fontSize": 14,
          },
          undefined,
        ]
      }
    />
    <View
      style={
        [
          {
            "flex": 1,
          },
          undefined,
        ]
      }
    />
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignSelf": "center",
          "backgroundColor": "transparent",
          "borderColor": "#2E50D4",
          "borderRadius": 8,
          "borderStyle": "solid",
          "borderWidth": 1,
          "height": 52,
          "justifyContent": "center",
          "opacity": 1,
          "padding": 8,
          "width": "100%",
        }
      }
    >
      <Text
        style={
          [
            {
              "color": "#2E50D4",
              "fontFamily": "Nunito-Bold",
              "fontSize": 14,
              "textAlign": "center",
            },
            undefined,
          ]
        }
        variant="outlined"
      >
        Sair
      </Text>
    </View>
  </View>
</RCTSafeAreaView>
`;
